website/blog/2018-01-10-1.10.0.md

```diff
@@ -0,0 +1,411 @@
+
+Prettier's in-built languages have been refactored to be expressed in terms of the plugin API, this way we can guarantee going forward that the core API is generic.
+
+Check [the docs][plugin api] for more information.
+
+> Please note that as this is a new, significantly large feature, we're releasing this under a **beta** label in the documentation. While we don't anticipate anything significant, this means that we may make breaking changes in the next release.
+
+### Flow
+
+#### Print flow type annotation comments as comments ([#3449]) by [@duailibe]
+
+Flow [type annotation comments](https://flow.org/en/docs/types/comments/#toc-comment-type-annotation) are a nice way to get type checking without having to transpile your files. But before this release, Prettier would not print the type annotation as comments, making it impossible to use that feature; now Prettier will correctly detect if the type annotation was a comment and print accordingly.
+
+```ts
+// Input
+let foo /*: string */ = "a";
+
+// Before
+let foo: string = "a";
+
+// After
+let foo /*: string */ = "a";
+```
+
+## Other Changes
+
+### JavaScript
+
+#### Print comment after arrow function params ([#3444]) by [@duailibe]
+
+<!-- prettier-ignore -->
+```ts
+// Before
+const run = (cmd /*: string */ /*: Promise<void> */) => {};
+
+// After
+const run = (cmd /*: string */) /*: Promise<void> */ => {};
+```
+
+### TypeScript
+
+### Support numeric separators ([#3580]) by [@azz]
+
+<!-- prettier-ignore -->
+```ts
+// Before
+SyntaxError: ',' expected. (1:10)
+> 1 | var a = 1_000_000_000;
+    |          ^
+  2 | var b = 0b1101_0101_1001;
+  3 | var c = 0xAE_FE_2F;
+
+// After
+var a = 1_000_000_000;
+var b = 0b1101_0101_1001;
+var c = 0xAE_FE_2F;
+```
+
+### Flow
+
+#### Print parens in FunctionTypeAnnotation when arrowParens is "always" ([#3616]) by [@duailibe]
+
+<!-- prettier-ignore -->
+```ts
+// --arrow-parens always
+
+// Before
+type SomeType = {
+  something: number => number
+};
+
+// After
+type SomeType = {
+  something: (number) => number
+};
+```
+
+### JSX
+
+#### Inline do expressions inside of JSX ([#3607]) by [@vjeux]
+
+<!-- prettier-ignore -->
+```jsx
+// Before
+{
+  do {
+    // ...
+  }
+}
+
+// After
+{do {
+ // ...
+}}
+```
+
+#### Prevent adding softline after arrow attribute with comments ([#3641]) by [@duailibe]
+
+<!-- prettier-ignore -->
+```jsx
+// --print-width 13 (for demonstration)
+
+// Before
+<span
+  attr={
+    // comment
+    () =>
+      true
+
+  }
+/>;
+
+// After
+<span
+  attr={
+    // comment
+    () =>
+      true
+  }
+/>;
+```
+
+#### Don't wrap JSX elements as attribute in `()` ([#3640]) by [@duailibe]
+
+The [JSX spec](http://facebook.github.io/jsx/) allows passing an element as an attribute to another element and Prettier was adding parens around, causing a syntax error. This was fixed in this release.
```

Comment:
You can mention that even though the jsx spec allows it, no current transpiler actually prints it correctly. Babel will with the v7